### ISSUE TITLE
[no-unnecessary-boolean-literal-compare] Incorrect fix when reversed

### ISSUE THREAD

**Issue by melusc** [typescript-eslint/typescript-eslint#3442]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

.eslintrc.json
```JSON
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "plugins": ["@typescript-eslint"],
  "rules": {
    "@typescript-eslint/no-unnecessary-boolean-literal-compare": "error"
  }
}

```

file.ts
```TS
const isOdd = (n: number): boolean => (n & 1) === 1;

if (false === isOdd(1)) {}
//  ^ "yoda way" of comparison
```

tsconfig.json
```json
{}
```

**Expected Result**

```TS
const isOdd = (n: number): boolean => (n & 1) === 1;

if (!isOdd(1)) {}
```

**Actual Result**

```TS
const isOdd = (n: number): boolean => (n & 1) === 1;

if (!false) {}
```

**Additional Info**

It behaves the same way with 4.25.1-alpha.0, 4.25.0, and 4.24.0.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.25.0` / `4.24.0` / `4.25.1-alpha.0` |
| `@typescript-eslint/parser`        | `4.25.0` |
| `TypeScript`                       | `4.2.4` |
| `ESLint`                           | `7.27.0` |
| `node`                             | `16.2.0` |
